### PR TITLE
haskellPackages.gitit: jailbreak

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Should fix the build failure described in #107209 for master/unstable.

The bug however is about NixOS 20.09, so my question is: Is it possible to backport haskell-updates fixes to stable? If yes how?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1171]: Instead of using `markUnbroken` here, can you remove `gitit` from the list of broken packages in `configuration-hackage2nix.yaml`?  The `hackage-packages.nix` is regenerated everyday automatically, so as long as `gitit` is not in the list of broken packages, it will be regenerated without `broken = true;`.

Also, if you could link to and upstream issue about the incorrect version constraints, that would be helpful (so that the Haskell maintainers here in nixpkgs will know when the `doJailbreak` can be removed).  If this is already fixed upstream (and a release just needs to be made), then just leave the expected version number in a comment above this line about when you expect the `doJailbreak` can be removed.

**Review by cdepillabout — CHANGES_REQUESTED** [review]
> Is it possible to backport haskell-updates fixes to stable? If yes how?

Sort of.  When backporting changes, you'll also have to directly edit the `hackage-packages.nix` file, since it is not automatically regenerated on the release branches.

Feel free to also open a PR for that.  You don't necessarily have to wait for this PR to get merged.

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix]: Done!

**Comment by cdepillabout** [comment]
Looks good, compiles for me.  Thanks!

**Pull request merged**
